lib/Doctrine/DBAL/Platforms/SQLAnywherePlatform.php

```diff
@@ -1121,10 +1121,26 @@ public function getTimeTypeDeclarationSQL(array $fieldDeclaration)
     /**
      * {@inheritdoc}
      */
-    public function getTrimExpression($str, $pos = TrimMode::UNSPECIFIED, $char = false)
+    public function getTrimExpression(string $str, int $mode = TrimMode::UNSPECIFIED, ?string $char = null) : string
     {
-        if (! $char) {
-            switch ($pos) {
+        switch ($mode) {
```

Comment:
I'd change this to `if (! in_array(...)) {`

Reply:
BTW: thoughts on including an assertion library for this stuff, @doctrine/doctrinecore?

Reply:
In this specific case, it'd reduce code duplication but wouldn't reduce knowledge duplication. Feels like sweeping mess under the rug.